registry/providermirrorprotocol/openapi.yml

```diff
@@ -0,0 +1,72 @@
+definitions:
+    ListAvailableInstallationPackagesArchive:
+        properties:
+            hashes:
+                description: |-
+                    A list of hashes for the ZIP archive.
+
+                    TODO: describe the hash format.
```

Comment:
This is certainly not something we need to treat as a blocker for merging the client code here :grinning: but I'd separately like to factor out the calculation functions for both of the currently-supported hash schemes (the `h1:` and `zh:` schemes) into some other library.

It seems beyond the scope of _this_ library, which is focused on the registry protocols themselves rather than the specific data within, but perhaps there's something in a separate library that knows how to work with the actual provider packages (both in their `.zip` archive and unpacked forms), including generating checksums for them, and then that could be shared between [OpenTofu's `package getproviders`](https://pkg.go.dev/github.com/opentofu/opentofu/internal/getproviders) and separate tools such as Renovate updating a dependency lock file, building a provider mirror with more flexibility than `tofu providers mirror` offers today, etc.

(This comment is really just notes we might refer back to in future, not actually feedback on your PR. Feel free to immediately resolve it. :grinning: )